compiler/x/codegen/UnaryEvaluator.cpp

```diff
@@ -36,6 +36,32 @@
 
 extern TR::Register *intOrLongClobberEvaluate(TR::Node *node, bool nodeIs64Bit, TR::CodeGenerator *cg);
 
+TR::Register *OMR::X86::TreeEvaluator::unaryVectorArithmeticEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *valueNode = node->getChild(0);
+   TR::Register *resultReg = cg->allocateRegister(TR_VRF);
+   TR::Register *valueReg = cg->evaluate(valueNode);
```

Comment:
Rather than always allocating into a register the common pattern most evaluators follow for instructions that have source memory forms is something like this:
```
   // If the child's register is NULL that means it hasn't been evaluated before (i.e., this is
   // the first occurrence).  And if the reference count is 1 that means this is the only
   // occurrence.  If that's the case, then access the child directly from memory using a
   // Mem form instruction.
   //
   if ( (valueNode->getRegister() == NULL) && (valueNode->getReferenceCount() == 1) )
      {
      generateRegMemInstruction(OMR::InstOpCode::SQRTPDRegMem, node, resultReg, generateX86MemoryReference(valueNode, cg), cg);
      }
   else
      {
      generateRegRegInstruction(OMR::InstOpCode::SQRTPDRegReg, node, resultReg, valueReg, cg);
      }
```

Reply:
sqrtpd seems to require a 16-byte aligned memory operand. I say 'seems' because I couldn't find any documentation on this, but aligning the input in the tril test was the only way I could get the test to pass on windows.